core/src/main/scala/org/apache/spark/SparkEnv.scala

```diff
@@ -76,7 +76,8 @@ class SparkEnv (
 
   // A general, soft-reference map for metadata needed during HadoopRDD split computation
   // (e.g., HadoopFileRDD uses this to cache JobConfs and InputFormats).
-  private[spark] val hadoopJobMetadata = new MapMaker().softValues().makeMap[String, Any]()
+  private[spark] val hadoopJobMetadata =
```

Comment:
This object was really a cache anyway and it was recommended to use CacheBuilder instead of MapMaker.